test/errors.spec.js

```diff
@@ -0,0 +1,73 @@
+'use strict'
+
+const Hapi = require('hapi')
+const Joi = require('joi')
+const should = require('should')
+
+let server, buildServer, destroyServer, hh
+
+const schema = {
+    type: 'brands',
+    attributes: {
+        code: Joi.string().min(2).max(10),
+        year: Joi.number(),
+        series: Joi.number(),
+        description: Joi.string()
+    }
+}
+
+const data = {
+    type: 'brands',
+    attributes: {
+        code: 'MF',
+        year: 2007,
+        series: 5,
+        description: 'Massey Furgeson'
+    }
+}
+
+describe('Global Error Handling', function () {
+	beforeEach(function (done) {
+		buildServer(() => {
+			done();
+		})
+	})
+
+    afterEach(function (done) {
+        destroyServer(done)
+    })
+
+	describe('Given a request for an invalid resource', function () {
+		it('should return a JSON+API compliant error', function () {
+			return server.injectThen({ method: 'get', url: '/some/bogus/request'})
+			.then(res => {
+				res.statusCode.should.equal(404)
+				expect(res.headers['content-type']).to.equal('application/vnd.api+json')
+                should.exist(res.payload)
+                let payload = JSON.parse(res.payload)
+                payload.should.have.property('errors').and.be.an.Array
```

Comment:
I used multiple assertions here since it's easier to zero in on the error when it fails.  Also some of these properties are optional, and may or may not exist in the future. It will be easier to make them optional/flexible in the future if it's not one big and specific object we're checking against.